services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from http import HTTPStatus
+from typing import Any, Literal, Mapping, Optional, Tuple
+
+from libcommon.constants import PROCESSING_STEP_CONFIG_OPT_IN_OUT_URLS_SCAN_VERSION
+from libcommon.simple_cache import DoesNotExist, SplitFullName, get_response
+
+from worker.job_runner import (
+    JobResult,
+    JobRunner,
+    JobRunnerError,
+    ParameterMissingError,
+    get_previous_step_or_raise,
+)
+from worker.utils import OptInOutUrlsScanResponse
+
+ConfigOptInOutUrlsScanJobRunnerErrorCode = Literal["PreviousStepFormatError"]
+
+
+class ConfigOptInOutUrlsScanJobRunnerError(JobRunnerError):
+    """Base class for exceptions in this module."""
+
+    def __init__(
+        self,
+        message: str,
+        status_code: HTTPStatus,
+        code: ConfigOptInOutUrlsScanJobRunnerErrorCode,
+        cause: Optional[BaseException] = None,
+        disclose_cause: bool = False,
+    ):
+        super().__init__(
+            message=message, status_code=status_code, code=code, cause=cause, disclose_cause=disclose_cause
+        )
+
+
+class PreviousStepFormatError(ConfigOptInOutUrlsScanJobRunnerError):
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+def compute_opt_in_out_urls_scan_response(dataset: str, config: str) -> Tuple[OptInOutUrlsScanResponse, float]:
+    logging.info(f"get config-opt-in-out-urls-scan for dataset={dataset} config={config}")
+
+    split_names_response = get_previous_step_or_raise(
+        kinds=["/split-names-from-streaming"], dataset=dataset, config=config
```

Comment:
we depend on /split-names-from-streaming, we should add it to the requires of the step

Reply:
Fixed in https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/libs/libcommon/src/libcommon/config.py#L283